File: Atividade_9/imprimir.go
package main

import "fmt"


type Imprimivel interface {
	imprimir()
}


type Relatorio struct {
	nome     string
	ano      int
	conteudo string
}

func (r Relatorio) imprimir() {
	fmt.Printf("Imprimindo Relatório: %s (%d)\n", r.nome, r.ano)
	fmt.Println(r.conteudo)
}


type Contrato struct {
	nome     string
	ano      int
	conteudo string
}

func (c Contrato) imprimir() {
	fmt.Printf("Imprimindo Contrato: %s (%d)\n", c.nome, c.ano)
	fmt.Println(c.conteudo)
}

func main() {
	
	relatorio := Relatorio{
		nome:     "Relatório de Vendas",
		ano:      2024,
		conteudo: "Conteúdo do Relatório de Vendas. -444-324",
	}

	
	contrato := Contrato{
		nome:     "Contrato de Prestação de Serviços",
		ano:      2024,
		conteudo: "Conteúdo do Contrato. Contrato feito por ambas as partes.",
	}

	
	relatorio.imprimir()
	contrato.imprimir()
}

File: Atividade_9/imprimir.py
from abc import ABC, abstractmethod


class Imprimivel(ABC):
    @abstractmethod
    def imprimir(self):
        pass


class Relatorio(Imprimivel):
    def __init__(self, nome, ano, conteudo):
        self.nome = nome
        self.ano = ano
        self.conteudo = conteudo

    def imprimir(self):
        print(f"Imprimindo Relatório: {self.nome} ({self.ano})")
        print(self.conteudo)


class Contrato(Imprimivel):
    def __init__(self, nome, ano, conteudo):
        self.nome = nome
        self.ano = ano
        self.conteudo = conteudo

    def imprimir(self):
        print(f"Imprimindo Contrato: {self.nome} ({self.ano})")
        print(self.conteudo)


if __name__ == "__main__":
    
    relatorio = Relatorio("Relatório de Vendas", 2024, "Conteúdo do Relatório de Vendas. -444-324")

    
    contrato = Contrato("Contrato de Prestação de Serviços", 2024, "Conteúdo do Contrato. Contrato feito por ambas as partes.")

    
    relatorio.imprimir()
    contrato.imprimir()
